crates/ruff_linter/src/rules/pylint/rules/bad_super_call.rs

```diff
@@ -0,0 +1,130 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{
+    Expr, ExprAttribute, ExprCall, ExprName, Stmt, StmtClassDef, StmtExpr, StmtFunctionDef,
+};
+
+use crate::checkers::ast::Checker;
+
+/// ### What it does
+/// Checks to see another argument other than the current class is given as the first argument of the super builtin
+///
+/// ### Why is this bad?
+/// In Python 2.7, `super()` has to be called with its own class and `self`
+/// as arguments `(super(Cat, self))`, which can lead to a mix up of parent and child class in the code.
+///
+/// In Python 3 the recommended way is to call `super()` without arguments
+/// (see also [`super-with-arguments`](https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/super-with-arguments.html)).
```

Comment:
What are the use cases when I would use this rule over `super-with-arguments` when targeting Python 3+ (the only version ruff supports)?

Reply:
I don't think you would to be honest, `super-with-arguments` is also a "suggestion".